applications/GeoMechanicsApplication/custom_workflows/time_loop_executor.hpp

```diff
@@ -96,7 +102,14 @@ public :
         return end_state;
     }
 
+    bool IsCancelled() const
+    {
+        return mCancelDelegateAvailable && mCancelDelegate();
```

Comment:
I was just thinking and discussing about this extra mCancelDelegateAvailable, and we don't need it. What we can do here is:
```suggestion
        return mCancelDelegate && mCancelDelegate();
```
if you call mCancelDelegate, it will return false when it is not set, so the order is important here.